### PR TITLE
Hipify config

### PR DESCRIPTION
**Pull request by bmedishe**
updated hipify_cli.py to take arguments from json and built warp_transducer(http://gitlab1.amd.com/asmathew/warp-transducer/-/tree/use_hipify_config)  successfully

### CODE REVIEW

**Comment by bmedishe** [comment]
@pruthvistony  could you check if this PR could be merged to master

File: cmake/Hipify.cmake [hunk 1]
```diff
@@ -79,8 +79,7 @@ function(hipify)
   if(HIPIFY_CONFIG_FILE)
     set(HIPIFY_COMMAND
       ${HIPIFY_DIR}/hipify_cli.py
-      --config-file ${HIPIFY_CONFIG_FILE}
-      --dump-dict-file ${HIPIFY_DICT_FILE}
```
Comment (pruthvistony) [inline cmake/Hipify.cmake:83]: set a default value for dump-dict-file in json file scenario.

File: hipify_config.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (pruthvistony) [inline hipify_config.json:1]: Why is this added into hipify repo?

Is the intention to add a sample file, then I would suggest to put into README.md or documentation.

File: hipify_config.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+    "project_directory": "/root/warp-transducer",
+    "output_directory" : "/root/warp-transducer",
+    "header_include_dirs": ["include"],
+    "includes":["src/*","test/*", "pytorch_binding/*", "tensorflow_binding/*", "include/*" ],
+    "dump_dict_file":""
```
Comment (pruthvistony) [inline hipify_config.json:6]: Need a default value as suggested before.

**Review by pruthvistony — CHANGES_REQUESTED** [review]

File: hipify_config.json (hunk 2 above)
Comment (jithunnair-amd) [reply hipify_config.json:1]: @bmedishe I agree, let's remove this file and add the sample contents to the README.md. Also, in the example, `project_directory` should be a relative path, preferably relative to the location of the hipify_config.json file to make it most intuitive to the user. Lmk if that doesn't work.

File: cmake/Hipify.cmake (hunk 1 above)
Comment (jithunnair-amd) [reply cmake/Hipify.cmake:83]: @pruthvistony This is tricky because the default value we want for dump-dict-file in the CMake hipify case seems to use a CMake variable: `set(HIPIFY_DICT_FILE ${CMAKE_BINARY_DIR}/hipify_output_dict_dump.txt)`. That means we can't have the default value in the json file exactly match the above CMake variable. Thoughts?

**Review by jithunnair-amd — CHANGES_REQUESTED** [review]

File: cmake/Hipify.cmake (hunk 1 above)
Comment (pruthvistony) [reply cmake/Hipify.cmake:83]: There are few points to take care and update the design accordingly.
1) dumping of hipify dict is a must to support get_hipified_list() API, which is an external API, which can be called from user after the hipify_cli.py trigger either through config file or specifying parameters.
2) We shouldnt control the dict dumping ON/OFF through json-config parameter [code-link](https://github.com/ROCmSoftwarePlatform/hipify_torch/pull/14/files/d20f2d94505a2124fbb7ac380ccfcb2b4a995754#diff-0b7bddf11afd4d0bda06f4377234786c0d9354f2c038e8dbd828dbba1837bcdaR94). So if we want configurability we can control only the name and have default value for 'HIPIFY_DICT_FILE'.

let me know your thoughts.
cc @jithunnair-amd @bmedishe

**Review by pruthvistony — CHANGES_REQUESTED** [review]

**Review by pruthvistony — CHANGES_REQUESTED** [review]
As discussed please update the handling of dump_dict_file.

File: hipify_cli.py [hunk 4]
```diff
@@ -57,23 +56,63 @@ def main():
         help="The file to Store the return dict output after hipification",
         required=False)
 
-    args = parser.parse_args()
-    print(args)
+    parser.add_argument(
+        '--config-json',
+        type=str,
+        default="",
```
Comment (jithunnair-amd) [inline hipify_cli.py:62]: Should this be defaulting to empty string? I think it shouldn't have a default, right?

File: hipify_cli.py [hunk 5]
```diff
@@ -57,23 +56,63 @@ def main():
...
+        help="The root of the project. (default: %(default)s)",
```
Comment (jithunnair-amd) [inline hipify_cli.py:63]: The help string is incorrect.

File: cmake/Hipify.cmake (hunk 1 above)
Comment (jithunnair-amd) [reply cmake/Hipify.cmake:83]: For the scope of this PR, I'm okay with having `dump-dict-file` as a separate argument to hipify_cli.py. But we should try to see if there's a way to unify all hipify_cli.py arguments into the json file while satisfying the requirements of the CMake hipify flow.

File: README.md [hunk 6]
```diff
@@ -54,7 +54,22 @@ Note: Update the CMAKE_MODULE_PATH list accordingly, if the hipify_torch repo is
...
+### hipify_cli.py 
+hipify_cli.py takes arguments through command line or a json file(hipify_config.json)
+```
+python hipify_cli.py --config-json hipify_config.json
+```
+hipify_config.json 
+
+{
+
+    "project_directory": <abs path to project>,
```
Comment (jithunnair-amd) [inline README.md:66]: This should be relative path, relative to config json.

**Review by pruthvistony — APPROVED** [review]

**Review by jithunnair-amd — APPROVED** [review]
